cirq/study/resolver.py

```diff
@@ -133,6 +131,8 @@ def __getitem__(self, key):
         return self.value_of(key)
 
     def __hash__(self):
+        if not hasattr(self, '_param_hash'):
```

Comment:
Might be better to set `self._param_hash = None` in `__init__` and then check for the `None` value here.

Reply:
Good idea.  Done.